homeassistant/components/life360/config_flow.py

```diff
@@ -1,108 +1,251 @@
+        self._password: str | vol.UNDEFINED = vol.UNDEFINED
+        self._options: dict[str, Any] = {}
+        self._reauth_entry: ConfigEntry | None = None
+        self._first_reauth_confirm = True
 
-        if user_input is not None:
-            self._username = user_input[CONF_USERNAME]
-            self._password = user_input[CONF_PASSWORD]
-            try:
-                # pylint: disable=no-value-for-parameter
-                vol.Email()(self._username)
-                authorization = await self.hass.async_add_executor_job(
-                    self._api.get_authorization, self._username, self._password
-                )
-            except vol.Invalid:
-                errors[CONF_USERNAME] = "invalid_username"
-            except LoginError:
-                errors["base"] = "invalid_auth"
-            except Life360Error as error:
-                _LOGGER.error(
-                    "Unexpected error communicating with Life360 server: %s", error
-                )
-                errors["base"] = "unknown"
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: ConfigEntry) -> Life360OptionsFlow:
+        """Get the options flow for this handler."""
+        return Life360OptionsFlow(config_entry)
+
+    async def _async_verify(self, step_id: str) -> FlowResult:
+        """Attempt to authorize the provided credentials."""
+
+        assert self._api
+        assert self._username
+        assert self._password
+
+        errors: dict[str, str] = {}
+        authorization = await get_life360_authorization(
+            self.hass, self._api, self._username, self._password, errors
+        )
+        if errors:
+            if step_id == "user":
+                schema = account_schema(
+                    self._username, self._password
+                ) | _account_options_schema(self._options)
             else:
-                if self._username in self.configured_usernames:
-                    errors["base"] = "already_configured"
-                else:
-                    return self.async_create_entry(
-                        title=self._username,
-                        data={
-                            CONF_USERNAME: self._username,
-                            CONF_PASSWORD: self._password,
-                            CONF_AUTHORIZATION: authorization,
-                        },
-                        description_placeholders={"docs_url": DOCS_URL},
-                    )
-
-        data_schema = OrderedDict()
-        data_schema[vol.Required(CONF_USERNAME, default=self._username)] = str
-        data_schema[vol.Required(CONF_PASSWORD, default=self._password)] = str
+                # Don't show current password the first time we prompt for password
+                # since this will happen asynchronously. However, once the user enters a
+                # password, we can show it in case it's not valid to make it easier to
+                # enter a long, complicated password.
```

Comment:
This is not a backend problem to solve. If we want the user to be able to show the password in the form that should be an option in the frontend.

Reply:
Ok. Don't know how to do that, but I'll get rid of this logic from the backend.